src/Packing.cpp

```diff
@@ -0,0 +1,67 @@
+//
+// Created by rolf on 03-03-21.
+//
+
+#include "Packing.h"
+#include "utilities.h"
+#include <cmath>
+
+RobotCommandPayload createLowLevelCommand(const proto::RobotCommand& proto,  const proto::World& world, bool isYellow){
+  float rho = sqrtf(proto.vel().x() * proto.vel().x() + proto.vel().y() * proto.vel().y());
+  float theta = atan2f(proto.vel().y(), proto.vel().x());
+  auto bot = rtt::robothub::utils::getWorldBot(proto.id(),isYellow,world);
+
+  RobotCommand command;
+  command.header = PACKET_TYPE_ROBOT_COMMAND;
+  command.id = proto.id();
+
+  command.doKick = proto.kicker();
+  command.doChip = proto.chipper();
+  command.doForce = proto.chip_kick_forced();
+
+  command.rho = rho;
+  command.theta = theta;
+  command.angle = proto.w();
+
+  if(bot){
+    command.useCameraAngle = true;
+    command.cameraAngle = bot->w();
+  }else{
+    command.useCameraAngle = false;
+    command.cameraAngle = 0.0;
+  }
+  command.kickChipPower = proto.chip_kick_vel();
+  command.dribbler = proto.dribbler();
+  command.angularControl = proto.use_angle();
+  command.feedback = false;
+
+  RobotCommandPayload payload;
+  encodeRobotCommand(&payload,&command);
+  return payload;
+}
```

Comment:
In general, for all variables : In what ranges do the values in the proto messages fall? 
We should add checks.

Reply:
Maybe I'm misunderstanding, but isn't the whole point of the roboteam_embedded_messages repository that these checks and unit conversions are done automatically? If anything, I don't think that's the responsibility of robothub.

One thing, which you could maybe do, is for each field with a bound add a RobotCommad_get_field_upperbound() and RobotCommand_get_field_lowerbound() functions for any field of the command. Since RobotHub is not about the conversions, I don't think it makes sense to have the unit conversion bound checks happening in the RobotHub code directly.

Reply:
You could even add a bool isValid() function to RobotCommand to validate that the units are in range.

Reply:
roboteam_embedded_messages doesn't do any checking at the moment. It simply assumes that all values are in the correct range. If a given value is too large, the resulting integer will overflow and a very low value would come out at the robot end.

I like the upperboud / lowerbound idea. Very simple to implement. I'll think about adding checks to roboteam_embedded_messages, but I really want to keep the repo as small as possible